### PR TITLE
Remote activity stub configuration issue

### PR DESCRIPTION
**Pull request by julia-elikhis**

### CODE REVIEW

**Comment by julia-elikhis** [comment]
I have opened this PR to suggest a fix to the following configuration issue:

When an activity interface does not have a local implementation, but - (1) there is a configuration for it in Temporal **properties** under `spring.temporal.activityStubs`, and (2) no configuration is provided on the **annotated** field of `@ActivityStub`- then `WorkflowAnnotationBeanPostProcessor` will fail to process the workflow bean. 

This happens because `WorkflowAnnotationBeanPostProcessor` assumes that all `ActivityStub` annotations with an empty `taskQueue` must have a local implementation. Which is not always true.
My solution is to add another check to `getBeansForInnerWorker`: `hasActivityStubConfiguration?`.

I've added a unit test and a sample application to verify the solution.
`RemoteActivityTest` mocks the activities and verifies that queue configuration is taken from `application.yaml`.
`RemoteActivity` runs an app that simulates the issue of bean creation (`WorkflowAnnotationBeanPostProcessor` won't run with test profile).
Also, since I needed to create an app to verify my solution, I fixed the issue of tests not being run when there are `@SpringBootApplication` annotations ("FIXME" comment).

File: src/test/java/ai/applica/spring/boot/starter/temporal/samples/BaseTest.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package ai.applica.spring.boot.starter.temporal.samples;
+
+import ai.applica.spring.boot.starter.temporal.annotations.TemporalTest;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@TemporalTest
+@SpringBootTest(classes = TestApplicationConfiguration.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+// @TestMethodOrder(MethodOrderer.MethodName.class)
```
Comment (julia-elikhis) [inline src/test/java/ai/applica/spring/boot/starter/temporal/samples/BaseTest.java:10]:
```suggestion
```

**Review by phntom — APPROVED** [review]

**Comment by phntom** [comment]
Would appreciate if this could get merged soon. Could a maintainer take look please?

**Comment by the-fine** [comment]
@simcha any chance to get this merged?

**Comment by simcha** [comment]
Sorry for the long reaction thanks for the changes.

**Pull request merged**